retrievalmarket/impl/provider.go

```diff
@@ -277,7 +285,7 @@ func (p *provider) NextBlock(ctx context.Context, id retrievalmarket.ProviderDea
 }
 
 func (p *provider) GetPieceSize(c cid.Cid) (uint64, error) {
-	pieceInfo, err := getPieceInfoFromCid(p.pieceStore, c)
+	pieceInfo, err := getPieceInfoFromCid(p.pieceStore, c, cid.Undef)
```

Comment:
This is called by ProposeDeal and DealProposal doesn't know anything about individual PieceCIDs. I'm assuming that client proposing a deal will always do a Query with the PieceCID first so unless PieceCID is added to DealProposal.Params,  we assume client has already determined the provider has what they want.